### PR TITLE
Behave graph cleanup

### PR DESCRIPTION
**Pull request by speigg**
Remove cyclic dependency between BehaveGraphComponent and BehaveGraphSystem modules. Cleanup some things.

### CODE REVIEW

File: packages/engine/src/behave-graph/components/BehaveGraphComponent.ts [hunk 1]
```diff
@@ -82,35 +82,22 @@ export const BehaveGraphComponent = defineComponent({
...
-        }
-      }))
-      systemState.get({ noproxy: true }).domains[domainName].register(registry)
-      setRegistry(registry)
-    }, [])
+    const canPlay = graph.run && !graph.disabled
+    const registry = behaveGraph.registries[graph.domain.value].get({ noproxy: true })
+    const graphRunner = useGraphRunner({ graphJson: graph.graph.get({ noproxy: true }), autoRun: canPlay, registry })
 
     useEffect(() => {
```
Comment (SYBIOTE) [inline packages/engine/src/behave-graph/components/BehaveGraphComponent.ts:92]: i think im breaking this up into two separate Useeffects

Comment (SYBIOTE) [reply packages/engine/src/behave-graph/components/BehaveGraphComponent.ts:92]: updated it , this is cleaner

**Review by SYBIOTE — APPROVED** [review]
this simplifies a lot of things and fuses the old and new code effectively

**Comment by speigg** [comment]
Still needs some cleanup

**Pull request merged**